### PR TITLE
Optimize docker image

### PR DESCRIPTION
**Pull request by lightnet328**
# Overview

See #445.
Thank you for creating issue easy to understand, @inductor.

# Changes

| | Commit | Size |
|:--|:--|:--|
| Before | | 1.43GB |
| Use Multi-stage build | c093783 | 729MB |
| Ignore .git | 8445930| 540MB |

These images are built for development.
In production, `mongodb-memory-server` is not installed. So the production image will be more lightweight.

### CODE REVIEW

**Comment by lightnet328** [comment]
memo: I want to upgrade Node.js version to 10.16 after this PR merged.
https://github.com/crowi/crowi/issues/410

**Comment by inductor** [comment]
Related issue: https://github.com/crowi/crowi/issues/445

File: .dockerignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 data
-node_modules
\ No newline at end of file
+node_modules
+.git
```
Comment (inductor) [inline .dockerignore]: Please put new line at EOF for POSIX compatibility.

http://hiroakiuno.hatenablog.com/entry/20070409/p1
https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

**Comment by inductor** [comment]
The image size became 530MB from 1.16GB ❤️

File: .dockerignore (hunk 1 above)
Comment (inductor) [reply .dockerignore]: @lightnet328 ping

Comment (lightnet328) [reply .dockerignore]: Thanks. I agree with that and fixed it.

**Pull request merged**